common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -143,6 +143,40 @@ public static boolean execICU(final UTF8String l, final UTF8String r,
    * Collation-aware regexp expressions.
    */
 
+  public static class StringSplit {
+    public static UTF8String[] exec(final UTF8String string, final UTF8String regex,
+        final int limit, final int collationId) {
+      CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
+      if (collation.supportsBinaryEquality) {
+        return execBinary(string, regex, limit);
+      } else {
+        return execLowercase(string, regex, limit);
```

Comment:
there is no `execICU` for this operation?

Reply:
If it's not supported yet, we should add `assert(collation.supportsLowercaseEquality)`

Reply:
I think `final lazy val collationId: Int = str.dataType.asInstanceOf[StringType].collationId` for `StringTypeBinaryLcase` won't allow that to happen. But an additional assert in the else branch couldn't hurt too